js/packages/openinference-instrumentation-langchain/src/instrumentation.ts

```diff
@@ -0,0 +1,112 @@
+import * as CallbackManagerModule from "@langchain/core/callbacks/manager";
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+  InstrumentationModuleDefinition,
+  InstrumentationNodeModuleDefinition,
+  isWrapped,
+} from "@opentelemetry/instrumentation";
+import { VERSION } from "./version";
+import { diag } from "@opentelemetry/api";
+import { LangChainTracer } from "./tracer";
+
+const MODULE_NAME = "@langchain/core/callbacks";
+
+export class LangChainInstrumentation extends InstrumentationBase<
+  typeof CallbackManagerModule
+> {
+  constructor(config?: InstrumentationConfig) {
+    super(
+      "@arizeai/openinference-instrumentation-langchain",
+      VERSION,
+      Object.assign({}, config),
+    );
+  }
+
+  protected init(): InstrumentationModuleDefinition<
+    typeof CallbackManagerModule
+  > {
+    const module = new InstrumentationNodeModuleDefinition<
+      typeof CallbackManagerModule
+    >(
+      "@langchain/core/dist/callbacks/manager.cjs",
+      ["^0.0.0", "^0.1.0"],
```

Comment:
why the list? doesn't the former sufice?

Reply:
I think we need both because of this left-most non zero element in the version, since the major version is 0

> Allows changes that do not modify the left-most non-zero element in the [major, minor, patch] tuple. In other words, this allows patch and minor updates for versions 1.0.0 and above, patch updates for versions 0.X >=0.1.0, and no updates for versions 0.0.X.

https://github.com/isaacs/node-semver?tab=readme-ov-file#caret-ranges-123-025-004

Reply:
hmmm, okay so we would have to update when they releasee 0.2 I guess?

Reply:
yes, i believe so